s3po/backends/s3.py

```diff
@@ -93,3 +93,10 @@ def func():
             multi.upload_part_from_file(StringIO(data), count)
             multi.complete_upload()
             return True
+
+    def list(self, bucket, prefix=None, delimiter=None, headers=None):
+        '''List the bucket, possibly limiting the search with a prefix.'''
+        bucket = self.conn.get_bucket(bucket)
+        # consume iterator to make a list to keep parity with Swift backend
+        results = list(bucket.list(prefix, delimiter, headers=headers))
+        return [result.name for result in results]
```

Comment:
This should probably be:

```python
# This returns a generator
return (key.name for key in bucket.list(prefix, delimiter, headers=headers))
```

If that's too long, the current form would be fine if you use `()` instead of `[]` on the `return` line.